features/dd-sdk-android-profiling/src/main/java/com/datadog/android/profiling/Profiling.kt

```diff
@@ -6,19 +6,27 @@
 
 package com.datadog.android.profiling
 
+import android.content.Context
 import android.os.Build
 import androidx.annotation.RequiresApi
 import com.datadog.android.Datadog
 import com.datadog.android.api.SdkCore
 import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.internal.time.DefaultTimeProvider
+import com.datadog.android.profiling.internal.NoOpProfiler
+import com.datadog.android.profiling.internal.Profiler
 import com.datadog.android.profiling.internal.ProfilingFeature
-import com.datadog.android.profiling.internal.perfetto.PerfettoProfilerProvider
+import com.datadog.android.profiling.internal.perfetto.PerfettoProfiler
+import com.datadog.android.profiling.internal.removeProfilingFlag
+import java.util.concurrent.Executors
 
 /**
  * An entry point to Datadog Profiling feature.
  */
 object Profiling {
 
+    private var profiler: Profiler = NoOpProfiler()
```

Comment:
this becomes a little bit fragile and probably won't scale for the future development, where we need to start/stop profiling manually and it will be bound to the particular SDK core.